### PR TITLE
Cleanup HealthChecker from migration logics

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/kind cleanup
/priority normal

**What this PR does / why we need it**:
This PR removes:
- the temporary logic for VPA controlplane health checks that allowed smooth migration (until a reconciliation with a new g/g version that adds the new labels) -> https://github.com/gardener/gardener/pull/2713
- the migration logic for the logging health checks that prevented health check to be executed for a Shoot cluster that is still not reconciled with g/g@v1.8.0 (that brings the loki based logging stack) -> https://github.com/gardener/gardener/pull/2515

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/health_check.go [hunk 1]
```diff
@@ -555,7 +543,7 @@ func (b *Botanist) checkControlPlane(
 		return exitCondition, err
 	}
 	if gardenletfeatures.FeatureGate.Enabled(features.Logging) {
-		if exitCondition, err := checker.CheckLoggingControlPlane(b.Shoot.SeedNamespace, b.isLoggingHealthCheckObsolete(), condition, seedStatefulSetLister); err != nil || exitCondition != nil {
+		if exitCondition, err := checker.CheckLoggingControlPlane(b.Shoot.SeedNamespace, b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting, condition, seedStatefulSetLister); err != nil || exitCondition != nil {
 			return exitCondition, err
 		}
 	}
```
Comment (vpnachev) [inline pkg/operation/botanist/health_check.go:549]: A few lines below, there is a temporary code about the metrics server health checking that can be removed in g/g@v1.14.x.

**Review by vpnachev — APPROVED** [review]
/lgtm


If you like, you can also clean up metrics server health check.

File: pkg/operation/botanist/health_check.go (hunk 1 above)
Comment (ialidzhikov) [reply pkg/operation/botanist/health_check.go:549]: Yeah, I also saw it but let's keep it for few more releases to make the gardener updates smooth (if someone from the community is updating several minor versions at once.)

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**